src/history.js

```diff
@@ -115,16 +116,36 @@ class History extends Emitter {
   /**
    * Map over the active branch.
    */
-  map(fn: (action: Action) => *, scope?: Object) {
-    let items = []
-    let cursor = this.root
+  // $FlowFixMe
+  [iteratorTag](): Iterator<Action, void> {
+    var cursor = this.root
 
-    while (cursor) {
-      items.push(fn.call(scope, cursor))
+    return {
+      next: () => {
+        var next = cursor
 
-      if (cursor == this.head) break
-      cursor = cursor.next
+        if (next == null) {
+          return { done: true }
+        }
+
+        cursor = next == this.head ? null : cursor.next
+
+        return { value: next, done: false }
+      }
```

Comment:
Here's a good overview:
https://babeljs.io/learn-es2015/#ecmascript-2015-features-iterators-for-of